test/customization_test.rb

```diff
@@ -0,0 +1,292 @@
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      **test(
+      ****int index, char character,
+      ****float precision, double trouble
+      **)
+    LANG
+  end
+
+  it 'must raise an error for a lambda of arity different than one' do
+    printer = Oppen::Wadler.new(margin: 45, space: ->(n, _o) { '*' * n })
+    _ { printer.output }.must_raise ArgumentError
+
+    printer = Oppen::Wadler.new(margin: 45, space: -> { '*' })
+    _ { printer.output }.must_raise ArgumentError
+
+    printer = Oppen::Wadler.new(margin: 45, space: ->(*_args) { '*' })
+    _ { printer.output }.must_raise ArgumentError
+  end
+end
+
+describe 'Line delimiter tests' do
+  it 'must work with a \n delimiter' do
+    printer = Oppen::Wadler.new(margin: 45, new_line: '\n')
+    customization_build_output(printer)
+    _(printer.output)
+      .must_equal 'function\n  test(\n    int index, char character,\n    float precision, double trouble\n  )'
+  end
+end
+
+describe 'Break delimiter tests' do
+  it 'must work with a different delimiter' do
+    printer = Oppen::Wadler.new(margin: 45)
+    printer.group {
+      printer.breakable('')
+      printer.text 'Hello'
+      printer.breakable(', ')
+      printer.text 'World!'
+    }
+    _(printer.output).must_equal 'Hello, World!'
+  end
+end
+
+describe 'Nest delimiter tests' do
+  def nest_delimiter_build_output(printer, open_obj, close_obj)
+    printer.group(0) {
+      printer.group(2) {
+        printer.text 'function'
+        printer.breakable
+        printer.text 'foo()'
+      }
+      printer.break
+      printer.nest(2, open_obj, close_obj) {
+        printer.group(0) {
+          printer.text 'Hello'
+          printer.breakable(', ')
+          printer.text 'World!'
+        }
+      }
+    }
+  end
+
+  it 'must work with an open and close object' do
+    printer = Oppen::Wadler.new(margin: 5)
+    nest_delimiter_build_output(printer, '{', '}')
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+        foo()
+      {
+        Hello
+        World!
+      }
+    LANG
+  end
+
+  it 'must work with no open and close object' do
+    printer = Oppen::Wadler.new(margin: 5)
+    nest_delimiter_build_output(printer, '', '')
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+        foo()
+      Hello
+        World!
+    LANG
+  end
+
+  it 'must work with only an open object' do
+    printer = Oppen::Wadler.new(margin: 5)
+    nest_delimiter_build_output(printer, '{', '')
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+        foo()
+      {
+        Hello
+        World!
+    LANG
+  end
+
+  it 'must work with only a close object' do
+    printer = Oppen::Wadler.new(margin: 5)
+    nest_delimiter_build_output(printer, '', '}')
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+        foo()
+      Hello
+        World!
+      }
+    LANG
+  end
+end
```

Comment:
We should also add a test on `nil` in left and right.

But should we raise an exception here, or simply coecre it to an empty space?
I personally dislike magical `nil`s …